File: backend/src/main/java/com/doyou/remember/backend/domain/Album.java
package com.doyou.remember.backend.domain;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
public class Album {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    
    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private Integer photoCount;

    @Column(name = "cover_image_id")
    private Long coverImageId;

    @Column(name = "cover_image_position")
    private String coverImagePosition;

    @ManyToMany
    @JoinTable(
        name = "album_files",
        joinColumns = @JoinColumn(name = "album_id"),
        inverseJoinColumns = @JoinColumn(name = "file_id")
    )
    private List<FileInfo> files = new ArrayList<>();

    @Column(name = "start_date")
    private LocalDateTime startDate;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
} 

File: frontend/src/app/types/album.ts
export interface Tag {
  id: number;
  name: string;
  createdAt: string;
}

export interface Album {
  id: number;
  title: string;
  description: string;
  photoCount: number;
  coverImageId?: number;
  coverImagePosition?: string;
  files: FileInfo[];
  startDate: string;
  createdAt: string;
  updatedAt: string;
  groupType?: 'date' | 'tag' | 'location';
  tags?: string[];
  location?: string;
}

export interface FileInfo {
  id: number;
  fileName: string;
  originalFileName: string;
  storagePath: string;
  fileSize: number;
  fileType: string;
  createdAt: string;
  tags?: Tag[];
}

export interface File {
  id: string;
  url: string;
  thumbnailUrl: string;
  storagePath: string;
  dateTaken: string;
  tags: string[];
  location?: string;
} 